p2p/libp2p/libp2pConnectionMonitor.go

```diff
@@ -35,6 +39,7 @@ func newLibp2pConnectionMonitor(reconnecter p2p.Reconnecter, thresholdMinConnect
 		thresholdDiscoveryResume:   0,
 		thresholdDiscoveryPause:    math.MaxInt32,
 		thresholdConnTrim:          math.MaxInt32,
+		sharder:                    &networksharding.NoSharder{},
```

Comment:
Will be refactored. As stated, the networkMessenger will have a predefined set of options that will be applied before running the ApplyOptions method on the networkMessenger. This will ensure the instance is ready to be used with some basic functionality activated that can be modified at a later time and will need a minimum provisioning of dependencies for that basic usage. The behavior is intensely used elsewhere, for example: https://github.com/libp2p/go-libp2p/blob/master/options.go